rust/src/lib.rs

```diff
@@ -1268,25 +1392,27 @@ pub unsafe extern "C" fn zcashlc_derive_transparent_private_key_from_seed(
         let network = parse_network(network_id)?;
         let seed = slice::from_raw_parts(seed, seed_len);
         let account = if account >= 0 {
-            account as u32
+            AccountId::from(account as u32)
         } else {
             return Err(format_err!("account argument must be positive"));
         };
 
-        let index = if index >= 0 {
+        let _index = if index >= 0 {
             index as u32
         } else {
             return Err(format_err!("index argument must be positive"));
         };
-        let sk = derive_secret_key_from_seed(&network, &seed, AccountId(account), index).unwrap();
-        let sk_wif = Wif::from_secret_key(&sk, true);
+        
+        let account_priv_key = extended_private_key_from_seed(&network, seed, account);
```

Comment:
Would it make sense to derive the unified spending key here instead, and then extract the transparent part?

Reply:
Oh, that's sound right. thanks

Reply:
the problem I have with this is that I need to serialize this and I can't serialize an AccountPrivKey.